## Resumenes tabulares
Con nuestra api de pandas exploramos nuestro conjunto de datos con las tablas de resumenes

In [1]:
%run 0.1-dataset.ipynb
%run pandas-API.ipynb

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload
The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload
The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [2]:
#Descripcion de las variables en general
datos.missing.missing_variable_summary()

variable  n_missing  n_cases  \
0                                 estoy_trabajando_en          0     5764   
1                              donde_estas_trabajando          0     5764   
2                                          dedicacion          0     5764   
3                                    tipo_de_contrato          0     5764   
4   ultimo_salario_mensual_o_retiro_bruto_en_tu_mo...          0     5764   
5   ultimo_salario_mensual_o_retiro_neto_en_tu_mon...        209     5764   
6                                    pagos_en_dolares       4215     5764   
7                 dolarizado_cual_fue_el_ultimo_valor       4781     5764   
8                              pagos_en_criptomonedas       5589     5764   
9     que_porcentaje_o_monto_recibis_en_criptomonedas       4974     5764   
10                         recibis_algun_tipo_de_bono          0     5764   
11                           a_que_esta_atado_el_bono          0     5764   
12  actualizaciones_de_tus_ingresos_laborales_dura...          0     5764   
13               de_que_fue_el_ajuste_total_acumulado        909     5764   
14                    en_que_mes_fue_el_ultimo_ajuste          0     5764   
15  ingresos_laborales_comparados_con_el_semestre_...          0     5764   
16                          con_que_beneficios_contas          0     5764   
17  que_tan_conforme_estas_con_tus_ingresos_laborales          0     5764   
18                             estas_buscando_trabajo          0     5764   
19                                         trabajo_de          0     5764   
20                                anos_de_experiencia          0     5764   
21                    antigüedad_en_la_empresa_actual          0     5764   
22                         tiempo_en_el_puesto_actual          0     5764   
23                     cuantas_personas_a_cargo_tenes          0     5764   
24       plataformas_que_utilizas_en_tu_puesto_actual          2     5764   
25  lenguajes_de_programacion_o_tecnologias_que_ut...          0     5764   
26    frameworksherramientas_y_librerias_que_utilices          1     5764   
27                                     bases_de_datos          1     5764   
28                                         qa_testing          5     5764   
29            cantidad_de_personas_en_tu_organizacion          0     5764   
30                               modalidad_de_trabajo          0     5764   
31   si_trabajas_hibrido_cuantos_dias_a_la_semana_vas          0     5764   
32     la_recomendas_como_un_buen_lugar_para_trabajar          0     5764   
33                         salir_o_seguir_contestando          0     5764   
34                           maximo_nivel_de_estudios       3111     5764   
35                                             estado       3111     5764   
36                                            carrera       3224     5764   
37                                        universidad       3715     5764   
38                    participaste_de_algun_boot_camp       5113     5764   
39  si_participaste_de_un_boot_campque_carrera_est...       5361     5764   
40      salir_o_seguir_contestando_sobre_las_guardias       3111     5764   
41                                     tenes_guardias       4025     5764   
42                          cuanto_cobras_por_guardia       4025     5764   
43                             porcentajebruto_o_neto       4025     5764   
44          salir_o_seguir_contestando_sobre_estudios       4025     5764   
45                                         tengo_edad          0     5764   
46                               me_identifico_genero          0     5764   
47                                  sueldo_dolarizado          0     5764   
48                                          seniority          0     5764   
49                                        me_id_extra        199     5764   

    pct_missing  
0      0.000000  
1      0.000000  
2      0.000000  
3      0.000000  
4      0.000000  
5      3.

In [3]:
# Descripcion de valores faltantes por variable
datos.missing.missing_variable_table()

n_missing_in_variable  n_variables  pct_variables
0                       0           28           56.0
1                    4025            4            8.0
2                    3111            3            6.0
3                       1            2            4.0
10                   3715            1            2.0
15                      2            1            2.0
14                      5            1            2.0
13                    199            1            2.0
12                    209            1            2.0
11                    909            1            2.0
8                    4215            1            2.0
9                    3224            1            2.0
7                    4781            1            2.0
6                    4974            1            2.0
5                    5113            1            2.0
4                    5361            1            2.0
16                   5589            1            2.0

Concluyendo: Contamos con 28 variables completas, que son el 56% de los datos totales, y 4 variables con 4025 variables valtantes

In [4]:
# Datos faltantes por fila
datos.missing.missing_case_summary()

case  n_missing  pct_missing
0        0         16    30.769231
1        1         15    28.846154
2        2         15    28.846154
3        3          6    11.538462
4        4         14    26.923077
...    ...        ...          ...
5759  5759          6    11.538462
5760  5760         15    28.846154
5761  5761         15    28.846154
5762  5762         15    28.846154
5763  5763         10    19.230769

[5764 rows x 3 columns]

In [5]:
datos.missing.missing_case_table()
#resumen de valores faltantes por fila
#nos muestra cuantas variables tiene n datos perdidos

n_missing_in_case  n_cases   pct_case
0                  15     1700  29.493407
1                   6      648  11.242193
2                  16      403   6.991672
3                  10      402   6.974323
4                  14      397   6.887578
5                  13      335   5.811936
6                   5      310   5.378210
7                   7      290   5.031228
8                  11      276   4.788341
9                   4      244   4.233171
10                  8      221   3.834143
11                  9      214   3.712700
12                 12      158   2.741152
13                  3       96   1.665510
14                  2       37   0.641915
15                 17       21   0.364330
16                  1        7   0.121443
17                 18        3   0.052047
18                  0        2   0.034698

In [6]:
# Por Intervalo de datos faltantes por bloque
(
    datos
    .missing
    .missing_variable_span(
        variable= "pagos_en_dolares",
        span_every=500
    )
)

span_counter  n_missing  n_complete  pct_missing  pct_complete
0              0        360         140     72.00000      28.00000
1              1        344         156     68.80000      31.20000
2              2        367         133     73.40000      26.60000
3              3        338         162     67.60000      32.40000
4              4        366         134     73.20000      26.80000
5              5        362         138     72.40000      27.60000
6              6        370         130     74.00000      26.00000
7              7        364         136     72.80000      27.20000
8              8        374         126     74.80000      25.20000
9              9        360         140     72.00000      28.00000
10            10        393         107     78.60000      21.40000
11            11        217          47     82.19697      17.80303

In [7]:
# Racha de valores faltantes por bloque
(
    datos
    .missing
    .missing_variable_run(
        variable= "participaste_de_algun_boot_camp",
    )
)

run_length     is_na
0              7   missing
1              1  complete
2              7   missing
3              1  complete
4              1   missing
...          ...       ...
1148          34   missing
1149           2  complete
1150          11   missing
1151           1  complete
1152          21   missing

[1153 rows x 2 columns]